src/librustc_errors/emitter.rs

```diff
@@ -749,6 +752,9 @@ fn overlaps(a1: &Annotation, a2: &Annotation) -> bool {
 fn emit_to_destination(rendered_buffer: &Vec<Vec<StyledString>>,
         lvl: &Level,
         dst: &mut Destination) -> io::Result<()> {
+    use lock;
+
+    let _buffer_lock = lock::acquire_global_lock("rustc_errors");
```

Comment:
I'll add a comment.

This section is only used for the new error format.  For the JSON output, this code won't be used.  JSON should already be output one message at a time.